dags/max_ai/run_evaluation.py

```diff
@@ -32,25 +39,56 @@ def get_object_storage_endpoint() -> str:
     return settings.OBJECT_STORAGE_ENDPOINT
 
 
-class ExportTeamsConfig(dagster.Config):
-    team_ids: list[int]
-    """Team IDs to run the evaluation for."""
+class PrepareDatasetConfig(dagster.Config):
+    dataset_id: str
+    """Dataset ID to run the evaluation for."""
+
+
+class PreparedDataset(BaseModel):
+    dataset_id: UUID
+    dataset_name: str
+    dataset_inputs: list[DatasetInput]
+
+
+@dagster.op(
+    description="Pulls the dataset and dataset items and validates inputs, outputs, metadata, and team_id presence in metadata."
+)
+def prepare_dataset(context: dagster.OpExecutionContext, config: PrepareDatasetConfig) -> PreparedDataset:
+    dataset = Dataset.objects.get(id=config.dataset_id)
+    dataset_items = DatasetItem.objects.filter(dataset=dataset).iterator(500)
+
+    dataset_inputs: list[DatasetInput] = []
+    try:
+        for dataset_item in dataset_items:
+            metadata = dataset_item.metadata or {}
+            dataset_inputs.append(
+                DatasetInput(
+                    input=dataset_item.input,
+                    expected=dataset_item.output,
+                    metadata=metadata,
+                    team_id=metadata.get("team_id"),
+                )
+            )
+    except ValidationError:
+        context.log.exception(f"Validation error for dataset item {dataset_item.id}")
```

Comment:
Nitpick: would it make sense to have try/except inside the loop, to ensure `dataset_item` always exists and always current?